lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -208,7 +201,102 @@ defmodule Trento.Domain.SapSystem do
         host_id: host_id,
         health: health
       }
-    ]
+    else
+      # TODO: Check error
+      {:error, :sap_system_not_registered}
+    end
+  end
+
+  # Sap system not registered, application already present
+  # If the istance is not one of MESSAGESERVER or ABAP we discard.
+  # Otherwise if the istance we want register together with already present istances
+  # have one MESSAGESERVER and one ABAP, we register the istance and the sap system
+  def execute(
```

Comment:
I know that this code is based in the old one, but I find it difficult to digest (even 2 nested levels of `if`s look to much now hehe).
I find easier the strategy we talked about the de-registration and using `multis` as needed.
So I would do:
- register application instance if it is not registered and if it is message or abap server (or always, if we decide to do so)
- in the next multi, check the sap system conditions, if abap and message server are present we register the sap system
- next multi we check if the health much be changed

This way, we don't need to have an almost duplicated new function maybe

Reply:
thanks, done :D